packages/go_router/doc/named-routes.md

```diff
@@ -0,0 +1,46 @@
+Instead of navigating to a route based on the URL, a GoRoute can be given a unique
+name. To configure a named route, use the `name` parameter:
+
+```dart
+GoRoute(
+   name: 'song',
+   path: 'songs/:songId',
+   builder: /* ... */,
+ ),
+```
+
+To navigate to a route using it's name, call [`goNamed`](https://pub.dev/documentation/go_router/latest/go_router/GoRouter/goNamed.html):
+
+```dart
+TextButton(
+  onPressed: () {
+    context.goNamed('song', params: {'songId': 123});
+  },
+  child: const Text('Go to song 2'),
+),
+```
+
+Alternatively, you can look up the location for a name using `namedLocation`:
+
+```dart
+TextButton(
+  onPressed: () {
+    final String location = context.namedLocation('song', params: {'songId': 123});
+    context.go(location);
+  },
+  child: const Text('Go to song 2'),
+),
+```
```

Comment:
Should we add a link back to the navigation topic? Something like: "Refer to the Navigation Topic to learn more about navigation." Same comment below for the redirection section.

Reply:
👍